builtin/bins/dkron-executor-rabbitmq/rabbitmq.go

```diff
@@ -32,64 +40,117 @@ func (r *RabbitMQ) Execute(args *dktypes.ExecuteRequest, cb dkplugin.StatusHelpe
 
 // ExecuteImpl do rabbitmq publish
 func (r *RabbitMQ) ExecuteImpl(args *dktypes.ExecuteRequest, cb dkplugin.StatusHelper) ([]byte, error) {
+	// validate config
+	cfg := args.Config
+	if cfg == nil {
+		return nil, errors.New("RabbitMQ config is empty")
+	}
+
+	url := cfg["url"]
+	if url == "" {
+		return nil, errors.New("RabbitMQ url is empty")
+	}
 
-	if args.Config["url"] == "" {
-		return nil, errors.New("url is empty")
+	queueName := cfg["queue.name"]
+	if queueName == "" {
+		return nil, errors.New("RabbitMQ queue name is empty")
 	}
 
-	if args.Config["queue"] == "" {
-		return nil, errors.New("queue is empty")
+	if cfg["message.body"] != "" && args.Config["message.base64"] != "" {
+		return nil, errors.New("RabbitMQ message.body and message.base64 are both set")
 	}
 
-	// broker := "amqp://guest:guest@localhost:5672/"
-	broker := args.Config["url"]
-	conn, err := amqp.Dial(broker)
+	// establish connection
+	conn, err := amqp.Dial(url)
 	if err != nil {
 		return nil, err
 	}
-	defer conn.Close()
-
+	defer func(conn *amqp.Connection) {
+		err := conn.Close()
+		if err != nil {
+			// DO NOTHING
+		}
+	}(conn)
 	ch, err := conn.Channel()
 	if err != nil {
 		return nil, err
 	}
-	defer ch.Close()
-
-	queue := args.Config["queue"]
-	q, err := ch.QueueDeclare(
-		queue, // name
-		false, // durable
-		false, // delete when unused
-		false, // exclusive
-		false, // no-wait
-		nil,   // arguments
-	)
-	if err != nil {
+	defer func(ch *amqp.Channel) {
+		err := ch.Close()
+		if err != nil {
+			// DO NOTHING
+		}
+	}(ch)
+
+	// create queue if necessary
+	if err := createQueueIfNecessary(args.Config, queueName, ch); err != nil {
+		return nil, err
+	}
+
+	// publish message
+	if err = publish(cfg, ch); err != nil {
 		return nil, err
 	}
+	return nil, nil
+}
+
+func createQueueIfNecessary(cfg map[string]string, queue string, ch *amqp.Channel) error {
+	if val, ok := cfg["queue.create"]; !ok || (ok && val == "false") {
```

Comment:
Should this query rabbit in order to check if the queue already exists?

Reply:
I think we don't need to check if the queue exists here. If the user chooses not to create the queue via the plugin, we expect him to create it manually. 
If the user doesn't declare the queue, message publishing will fail with error in logs or a failed execution in UI.